### PR TITLE
[KOGITO-7838] - Alignment of the API for the Workflow object

### PR DESCRIPTION
**Pull request by amadhusu**
**[KOGITO-7838]** - Alignment of the API for the Workflow object in the PoC with the Design document

Signed-off-by: Achyut Madhusudan <amadhusu@redhat.com>

**Description of the change:**

Alignment of the API for the Workflow object in the PoC with the Design document

**Motivation for the change:**

https://docs.google.com/document/d/1dvNlQ5tea2ixqOgucd-sLDPczrgwnYBNjbFyLkTvPUI/edit#heading=h.bmrte0aczgql

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

### CODE REVIEW

File: api/v08/kogitoserverlessworkflow_types.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package v08
 
 import (
+	"encoding/xml"
```
Comment (spolti) [inline api/v08/kogitoserverlessworkflow_types.go:20]: @davidesalerno shouldn't we be dropping these from here and use the sdk-go instead?

File: api/v08/kogitoserverlessworkflow_types.go [hunk 2]
```diff
@@ -225,11 +236,11 @@ type FunctionRef struct {
 	RefName      string            `json:"refName"`
 	Arguments    map[string]string `json:"arguments,omitempty"`
 	SelectionSet *string           `json:"selectionSet,omitempty"`
-	Invoke       *InvokeTye        `json:"invoke,omitempty"`
+	Invoke       *InvokeType       `json:"invoke,omitempty"`
 }
 
 type Action struct {
-	Name               *string           `json:"name"`
+	Name               *string           `json:"name,omitempty"`
```
Comment (spolti) [inline api/v08/kogitoserverlessworkflow_types.go:243]: on v08 name can be null, no need to be a pointer.

File: api/v08/kogitoserverlessworkflow_types.go (hunk 1 above)
Comment (spolti) [reply api/v08/kogitoserverlessworkflow_types.go:20]: before a full review, @davidesalerno let's align it.

Comment (amadhusu) [reply api/v08/kogitoserverlessworkflow_types.go:20]: @davidesalerno @spolti Anything I can do to help?

File: api/v08/kogitoserverlessworkflow_types.go (hunk 2 above)
Comment (amadhusu) [reply api/v08/kogitoserverlessworkflow_types.go:243]: @spolti Thanks for letting me know, i will change it during my next commit.

File: api/v08/kogitoserverlessworkflow_types.go (hunk 1 above)
Comment (davidesalerno) [inline api/v08/kogitoserverlessworkflow_types.go:20]: Why we need this import?

Comment (amadhusu) [reply api/v08/kogitoserverlessworkflow_types.go:20]: @davidesalerno I think my IDE settings pulled this in...I actually didn't import it...so will get that removed...

**Review by spolti — APPROVED** [review]
@davidesalerno it looks good.
But as we discussed yesterday, let's bring this up to discussion in the next meeting.

**Review by desmax74 — APPROVED** [review]

**Comment by davidesalerno** [comment]
> @davidesalerno it looks good. But as we discussed yesterday, let's bring this up to discussion in the next meeting.

+1

**Pull request merged**